### PR TITLE
V5sync

### PR DESCRIPTION
**Pull request by CIPop**
- Reuses MQTTClient as much as possible:
		○ I/O Model:
			§ Sync sending blocking functions that wait completion (ACK / PUBCOMP).
				□ Used for CONNECT/ACK, PUB/ACK, SUB/ACK, DISCONNECTsend
				□ Optionally, a "WithResults" version exists to retrieve data from the ACK
			§ Async receiving callbacks (must exit for the operation to complete)
				□ Used for PUBrecv, AUTH and DISCONNECTrecv
			
		○ #if defined(MQTTV5) used throughout MQTTClient.h/c
		○ One pending operation / call (blocking):
			§ CONNECT /w optional CONNACK information
			§ SUBSCRIBE (one topicFilter) /w optional SUBACK info
		
	- NEW: MQTTClient functionality:
		○ PUB /w optional PUBACK/PUBCOMP information (PubDoneData / MQTTPublishWithResults)

	- MQTTV5 Client differences:
		○ Ctor will require a reusable MQTTProperties* struct, allocated by the application.
		○ All APIs now accept or return MQTTProperties (CONNECT accepts both connect and will !?)
		○ New APIs, only available in V5:
			- MQTTV5Auth and MQTTV5SetAuthHandler (controlHandler)
			- MQTTV5SetDisconnectHandler (controlHandler)
			- MQTTV5UnsubscribeWithResults + MQTTV5UnsubackData

Open question:
- [ ] MQTTV5Packet - For symmetry, should MQTTPacket_willOptions (backed by MQTTPacket_willOptions_initializer) include the MQTTProperties* willProperties?

### CODE REVIEW

File: MQTTClient-C/src/V5/MQTTV5Client.h [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * @param readbuf 
+ * @param readbuf_size 
+ * @return DLLExport 
+ */
+DLLExport void MQTTV5ClientInit(MQTTClient* client, Network* network, unsigned int command_timeout_ms,
+		unsigned char* sendbuf, size_t sendbuf_size, unsigned char* readbuf, size_t readbuf_size, 
+    MQTTProperties* recvProperties);
+
+/** MQTT Connect - send an MQTT connect packet down the network and wait for a Connack
+ *  The nework object must be connected to the network endpoint before calling this
```
Comment (vaavva) [inline MQTTClient-C/src/V5/MQTTV5Client.h]:
```suggestion
 *  The network object must be connected to the network endpoint before calling this
```

File: MQTTClient-C/src/V5/MQTTV5Client.h [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * @brief Create an MQTT client object
+ * 
+ * @param client 
+ * @param network 
+ * @param command_timeout_ms 
+ * @param sendbuf 
+ * @param sendbuf_size 
+ * @param readbuf 
+ * @param readbuf_size
```
Comment (vaavva) [inline MQTTClient-C/src/V5/MQTTV5Client.h]:
```suggestion
 * @param readbuf_size 
 * @param recvProperties
```

Comment (vaavva) [reply MQTTClient-C/src/V5/MQTTV5Client.h]: Review params for this file - there are a few others missing as well

File: MQTTClient-C/src/V5/MQTTV5Client.h [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ *  @param client - the client object to use
+ *  @param topicFilter - the topic filter to subscribe to
+ *  @param message - the message to send
+ *  @param data - suback granted QoS returned
+ *  @return success code
+ */
+DLLExport int MQTTV5SubscribeWithResults(MQTTClient* client, const char* topicFilter, 
+  enum QoS qos, messageHandler messageHandler, MQTTSubackData* data);
+
+/** MQTT Subscribe - send an MQTT unsubscribe packet and wait for unsuback before returning.
```
Comment (vaavva) [inline MQTTClient-C/src/V5/MQTTV5Client.h]:
```suggestion
/** MQTT Unsubscribe - send an MQTT unsubscribe packet and wait for unsuback before returning.
```

File: MQTTClient-C/src/V5/MQTTV5Client.h (hunk 2 above)
Comment (CIPop) [reply MQTTClient-C/src/V5/MQTTV5Client.h]: Yes - documentation is not ready - sorry I forgot to add that limitation.

**Pull request closed without merging**